### PR TITLE
ci: Check gke cluster state before selecting it

### PR DESCRIPTION
**Pull request by nebril**
This change adds a check that confirms that GKE cluster is in running
state, which prevents build from failing when apiserver is already up,
but cluster is still being provisioned.

This kind of failure could be seen in this build: https://jenkins.cilium.io/job/Ginkgo-CI-Tests-Pipeline-GKE/2391/execution/node/112/log/

### CODE REVIEW

**Comment by nebril** [comment]
test-gke

**Comment by nebril** [comment]
Since this only affects gke cluster selection process, marking this as ready to merge.

File: test/gke/select-cluster.sh [hunk 1]
```diff
@@ -27,6 +27,13 @@ while [ $locked -ne 0 ]; do
 		continue
 	fi
 
+	echo "checking whether cluster ${cluster_uri} is running"
+	status=$(gcloud container clusters describe --project "${project}" --format="value(status)" "${cluster_uri}" --region="${region}")
+	if [[ "$status" != "RUNNING"  ]] ; then
+		echo "cluster status is $status , trying out other cluster"
```
Comment (christarazi) [inline test/gke/select-cluster.sh]: Nit: 
```suggestion
		echo "cluster status is $status , trying out another cluster"
```

**Review by christarazi — APPROVED** [review]
LGTM, one non-blocking nit.

**Pull request merged**